src/main/java/no/sikt/sws/IndexHandler.java

```diff
@@ -35,33 +40,52 @@ protected ProxyResponse<String> processProxyInput(
     ) throws ApiGatewayException {
 
         var resourceIdentifier = request.getPathParameter(RESOURCE_IDENTIFIER);
-
-        if (resourceIdentifier.startsWith("_") || !resourceIdentifier.matches(ALLOWED_INPUT)) {
-            throw new BadRequestException(
-                    "Root operations and indeces starting with '_' or containing anything but letters, digits, '/',"
-                            + " '-' or '_'are not allowed. Got: " + resourceIdentifier);
-        }
-
         var httpMethod = RequestUtil.getRequestHttpMethod(request);
         var workspace = RequestUtil.getWorkspace(request);
 
+        var searchCommand = OpenSearchCommand.fromString(resourceIdentifier);
+
+
         try {
-            var url = WorkspaceStripper.prefixUrl(resourceIdentifier, workspace);
+            if (searchCommand.equals(OpenSearchCommand.NOT_IMPLEMENTED)
+                || searchCommand.equals(OpenSearchCommand.INVALID)) {
 
+                validateResourceIdentifier(resourceIdentifier);
+            }
+
+            var url = "/" + WorkspaceStripper.prefixUrl(workspace, resourceIdentifier);
             logger.info("URL: " + url);
 
-            var response = openSearchClient.sendRequest(httpMethod, url, body);
+            if (body == null && (PUT ==  httpMethod || POST == httpMethod)) {
+                throw new IllegalArgumentException(REQUIRED_PARAMETER_IS_NULL + "[requestBody]");
+            }
+            var requestBody =
+                    (body != null) ? WorkspaceStripper.prefixBody(workspace, resourceIdentifier, body) : null;
+
+            var response = openSearchClient.sendRequest(httpMethod, url, requestBody);
 
             logger.info("response-code:" + response.getStatus());
             logger.info("raw response-body:" + response.getBody());
 
-            var responseBody = WorkspaceStripper.remove(response.getBody(), workspace);
+            var responseBody = WorkspaceStripper.removePrefix(workspace, response.getBody());
             logger.info("stripped response-body:" + responseBody);
 
             return new ProxyResponse<>(response.getStatus(), responseBody);
+        } catch (BadRequestException be) {
+            logger.error(be.getLocalizedMessage());
+            throw be;
```

Comment:
Since we are at the IndexHandler we should throw it as a managed exception so we know what is returned to the caller. Perhaps use throw new SearchException(e.getMessage(), e); or something similar

Reply:
ok